app/forms/idv/inherited_proofing/base_form.rb

```diff
@@ -35,6 +35,8 @@ def initialize(payload_hash:)
         raise ArgumentError, 'payload_hash is blank?' if payload_hash.blank?
         raise ArgumentError, 'payload_hash is not a Hash' unless payload_hash.is_a? Hash
 
+        super()
+
         self.class.attr_accessor(*self.class.fields)
```

Comment:
This code has already been approved, so perhaps we can leave this for a "tech debt" card if it comes down to it after all is said and done, but I think we should try to get this card in as soon as possible because I'm working on a card that id dependent upon this.

But:

> ...but why are we calling class methods in the initializer?
`attr_accessor` creation logic and data loading takes place in the base class. `.fields` that need to be created are defined in subclasses. This code resides in the base class initializer so that 1) subclasses only need to be concerned with field naming (not `attr` definition/creation) and 2) so that the `.fields` defined by the subclass are available to the base class to create them and load data into them. Defining the `attr_accessors` in the subclasses, I'd still have to provide some means (Array) for the base class to identify the `.fields` that need to be created and loaded with data.

> Best case scenario is this does nothing that couldn't be done in the class definition.
Not unless you move this code to the subclass or define them in the subclass, because `.fields` needs to be available for the base class to create the `attr_accessors` and load the data. But, to the point above: If the code to create the `attr_accessors` and load the data exists in the base class, I would still need to provide some Array to the base class to identify what `attr_accessors` to create/load; this is _part_ of what `required_fields`, `optional_fields` and `.fields` accomplishes. I could turn the base class into a mixin, but this comes with its own downsides, for example, the initializer would still have to be copied.

> ...worst case is it modifies this class while it is in use elsewhere.
There is no danger of modifying the class definition "while it is in use elsewhere" from what I see. This code relies on the `.fields` which is also a _class_ method, to create the `attr_accessors` - the fields can't be changed dynamically because they are defined at the class-level. If you don't like this because you believe it's redundant (re class-level modifications performed every instantiation), I can use `define_singleton_method w/ instance_variable_get/instance_variable_set`; in the end it's doing the same thing more or less, only on the singleton class, and more verbose.

Reply:
This code was already approved; comments above address the concerns. We can (if need be) address later.

Reply:
I also feel very strongly that the dynamically defined `self.class.attr_accessor(*self.class.fields)` is a very bad pattern. 

I understand that it was existing code, but clearly it slipped through our previous review passes.

If we don't clean it up in this PR, I need us to make a ticket to track cleaning it up ASAP as a follow-up

Reply:
Hi @zachmargolis, I can change this. Can you please elaborate on _why_, in this context, this "is a very bad pattern"?

Reply:
Addressed.